auth.php

```diff
@@ -1003,6 +1003,11 @@ public function logoutpage_hook() {
 
         // Only log out of the IdP if we logged in via the IdP. TODO check session timeouts.
         if ($auth->isAuthenticated()) {
+            
+            // Delete session cookie to address security concern in issue 499
+            $myCookie = 'MOODLEIDP1_'.$CFG->sessioncookie;
+            setcookie($myCookie, '', time() - HOURSECS, $CFG->sessioncookiepath, $CFG->sessioncookiedomain, $cookiesecure, $CFG->cookiehttponly);
```

Comment:
I believe $CFG and $cookiesecure are not defined here. Did you test the fix?

You can delete cookie by calling:
```
$this->set_idp_cookie('');
```